.github/workflows/test-builds-linux.yml

```diff
@@ -0,0 +1,102 @@
+name: Linux Test Builds
+
+on:
+  push:
+    branches: [ $default-branch ]
+  pull_request:
+    branches: [ $default-branch ]
+
+jobs:
+  style-check:
+    runs-on: ubuntu-16.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Check Style
+        run: ./ci/checksrc.sh
+  build:
+    runs-on: ubuntu-16.04
+    strategy:
+      fail-fast: false
+      matrix:
+        compiler: [gcc, clang]
+        address_size: [32, 64]
+        crypto_backend: [OpenSSL, Libgcrypt, mbedTLS]
+        build_shared_libs: [OFF, ON]
+        enable_zlib_compression: [OFF, ON]
+        b: [cmake]
+        include:
+          - compiler: gcc
+            address_size: 64
+            crypto_backend: OpenSSL
+            build_shared_libs: OFF
+            enable_zlib_compression: OFF
+            b: configure
+          - compiler: clang
+            address_size: 64
+            crypto_backend: OpenSSL
+            build_shared_libs: OFF
+            enable_zlib_compression: OFF
+            b: configure
+    env:
+      CC: ${{ matrix.compiler }}
+      CC_FOR_BUILD: ${{ matrix.compiler }}
```

Comment:
What is the purpose of this variable?

Reply:
I was matching Travis's `compiler` option behavior. They set both `CC` and `CC_FOR_BUILD` with the compiler selected. I don't think `CC_FOR_BUILD` is actually used anywhere, but I didn't want to take any chances.

https://docs.travis-ci.com/user/languages/c/#choosing-compilers-to-test-against

Reply:
https://www.linuxtopia.org/online_books/linux_development_tools/libtool_autoconf_automake_book/autobook_270.html

This explains what `CC_FOR_BUILD` is doing. I see no reason to keep it as we are not cross compiling, but at the same time it's doing no harm, so I'll leave the choice up to you.

Reply:
Okay, thanks for explaining this.